### PR TITLE
Build: Update dependencies, switch tests to ESM, JSHint -> ESLint

### PR DESCRIPTION
**Pull request by mgol**
Also, add a GitHub Actions workflow.

These updates are necessary for `node-packager` to work on newer Node.js - which, in turn, is needed for the jQuery UI release process.

### CODE REVIEW

**Comment by mgol** [comment]
Reviewing commit by commit might be easier since I isolated a commit with auto-generated changes from ESLint's autofix.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+[*]
+indent_style = tab
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.yml]
+indent_style = space
+indent_size = 2
```
Comment (timmywil) [inline .editorconfig:15]: FWIW, the package.json is currently using spaces.

**Review by timmywil — APPROVED** [review]
Looks good, but consider either changing JSON to use tabs or setting spaces in the `.editorconfig`.

File: .editorconfig (hunk 1 above)
Comment (mgol) [reply .editorconfig:15]: @timmywil It does not, I reformatted it in this PR

Comment (mgol) [reply .editorconfig:15]: Ah, there was a `test/package.json` file with 3 lines that used spaces for its single indent. Fixed now. But the main `package.json` I already updated before submitting the PR.

Comment (timmywil) [reply .editorconfig:15]: I think I had whitespace turned off in the diff

**Pull request merged**